docs/cli.md

```diff
@@ -174,31 +174,43 @@ For instance, for ES6 support with [Babel](https://babeljs.io/) 7 add:
 
 This will effectively call `require('@babel/register')` prior to requiring any support files.
 
-### Non JS files
+If your files end with an extension other than `js`, make sure to also include the `--require` option to state the required support files. For example, if using [CoffeeScript](https://www.npmjs.com/package/coffeescript):
 
-If your files end in an extension other than `js`, make sure to also include the `--require` option to state the support files to require.
+```
+--require-module coffeescript/register --require 'features/**/*.coffee'
+```
+
+### Typescript
 
-For example, with [TypeScript](https://www.typescriptlang.org/):
+#### With ts-node
+
+If you are using [ts-node](https://github.com/TypeStrong/ts-node):
 
 ```
 --require-module ts-node/register --require 'step-definitions/**/*.ts'
 ```
 
-or [CoffeeScript](https://www.npmjs.com/package/coffeescript):
+> ⚠️ Some TypeScript setups use `esnext` modules by default, 
+>   which doesn't marry well with Node. You may consider using commonjs instead.
+>   See how to add [extra configuration](#extra-configuration) below.
+
+#### With babel
+
+If you are using babel with [@babel/preset-typescript](https://babeljs.io/docs/en/babel-preset-typescript):
```

Comment:
Perhaps it would be good to add that for transpiling typescript with babel the user needs to have a `tests.setup.js` with the following: 
```js
// tests.setup.js
require("@babel/register")({extensions: [".ts"]})
```

and then a `.babelrc` file with at least:
```jsonc
// .babelrc
{ "presets": [ "@babel/preset-env", "@babel/preset-typescript"  ] }
```

Since if the '.ts' `extensions` @babel/register option is not set, it won't be run through @parcel/preset-typescript ([see](https://babeljs.io/docs/en/babel-preset-typescript)) 

And then we can have the options `--require test.setup.js --require 'src/**/*.steps.ts'` set.
At least that was the only way I managed to get it working with babel instead of ts-node.

Reply:
There is still plenty of room for improvements

Feel free to submit another PR as that one has been merged.

Myself, I would actually greatly encourage using ts-node instead of babel. To execute the tests, there is no need for too much transiplation.

Reply:
Definitely, let me prepare some doc updates and send for review. Not sure if you tried but this is also blazing fast `cucumber-js --require-module ts-eager/register --require features/support/*.ts`. [It's](https://github.com/mhart/ts-eager) the same as ts-node but transforms with esbuild :) So I might add a note for these recipes as well to provide some ideas of usage, if you think that'd be OK?